handlers/meme.js

```diff
@@ -0,0 +1,110 @@
+    });
+
+  // Before we make the macro, we need to query the API and find a generator to use.
+  //     The meme "generator" is actually the background image that is used for the macro.
+  //     We query the API for the current top 25 most popular generator images (25 is the max for the results page size),
+  //     and then randomly choose one of those to use.
+  let generatorOptions = {
+    uri: `http://version1.api.memegenerator.net//Generators_Select_ByPopular?pageIndex=0&pageSize=25&days=&apiKey=${MEMEGENERATOR_API_KEY}`,
+    json: true
+  };
+  rp.get(generatorOptions).then(generators => {
+    generatorID = _.sample(generators.result).generatorID;
+
+    // Now that we've gotten a generator, we can send our options to the meme forge and build our dark creation
+    let uri = `${API_URL}?apiKey=${MEMEGENERATOR_API_KEY}&generatorID=${generatorID}&languageCode=${languageCode}&text0=${top_text}&text1=${bottom_text}&username=${MEME_USER}&password=${MEME_PASSWORD}`;
+    let options = {
+      uri: uri,
+      json: true
+    };
+    rp.get(options)
+      .then(meme => {
+        // The API will spit back a bunch of stuff, namely the URL of the macro it just made
+        channel.send(meme.result.instanceImageUrl);
+      })
+      .catch(error =>{
+        // This will catch when we are able to query the generator, but the macro creation fails for some reason
+        //     As a fun treat, we can still send the top and bottom text of our meme-to-be to the channel
+        LOG.error(error);
+        channel.send("Encountered an error while connecting to the world meme database; we've been set up!");
+        channel.send(_.toUpper(top_text));
+        channel.send(_.toUpper(bottom_text));
+      });
+  })
+  .catch(error => {
+    // The memegenerator API is not reliable, and goes down often enough for it to be a nuisance. This displays when we can't connect
+    //     to the generator; usually after the server process times out. Again, we still have messages so we can still send those
+    LOG.error(error);
+    channel.send("Error retrieving meme generators from His Excellency, President for Life, Field Marshal Gaylord K. Memelord, VC, DSO, MC, Eternal Ruler of Heaven, Earth and the Interwebz, and All Creatures Who Crawl Swim and Fly Upon It, Past, Present and Future, In This and Any Other Dimension, Conqueror of the British Empire in Africa in General and Uganda in Particular, DDS");
+    channel.send(_.toUpper(top_text));
+    channel.send(_.toUpper(bottom_text));
+  });
+}
+
+// From the returned messages from discord, we will randomly choose one of those
+function findMessage(channel, messages){
+  let found = messages.random();
+  // If the returned message was sent by the bot or is a bot command, let's just recurse and try again until we get it right
+  if(found.author.id == channel.client.user.id || found.content.startsWith('.')){
+    found = findMessage(channel, messages);
```

Comment:
Recursion is cool, but not really necessary here.

Much better to use a while loop so we avoid a possible stack explosion haha